tests/test_config_model.py

```diff
@@ -11,6 +11,45 @@ def test_pydantic_model(self):
             CONFIG_MODEL.zones["US-NW-PSCO"].parsers.get_function("production")
         )
 
+    # Add well-known sources that don't require config-based references here
+    GLOBAL_SOURCE_REFERENCES = {
+        "2018 average estimated from https://www.hops.hr/page-file/oEvvKj779KAhmQg10Gezt2/temeljni-podaci/Temeljni%20podaci%202018.pdf",
+        "2020 average by Electricity Maps. See disclaimer",
+        "BEIS 2021",
+        "CEA 2022",
+        "EIA 2020/BEIS 2021",
+        "Enerdata 2022",
+        "EU-ETS 2021",
+        "EU-ETS, ENTSO-E 2021",
+        "Guatemala AMM 2021-2022",
+        "IEA 2019",
+        "IEA 2020",
+        'Oberschelp, Christopher, et al. "Global emission hotspots of coal power generation."',
+        "Tidal (IPCC 2014)",
+        "https://www.iea.org/data-and-statistics/charts/electricity-generation-mix-in-mexico-1-jan-30-sep-2019-and-2020",
+    }
+
+    def test_zone_sources(self):
```

Comment:
@pierresegonne following on from [my question yesterday](https://github.com/electricitymaps/electricitymaps-contrib/pull/4435#discussion_r1354593945): this test case is the reason I was wondering about allowing the `emissionFactors` key to remain on the in-memory zone config representations.

These changes are work-in-progress; defining a set of well-known `sources` within the code doesn't seem ideal, so I'm planning some follow-up these changes with a migration of `GLOBAL_SOURCE_REFERENCES` into another, new config YAML file (`config/sources.yaml`, or similar).

Also note: given your mention of memory usage as a concern: the changes here will require more memory not only for the loaded-YAML object representation of `emissionFactors`, but then another bunch of memory allocations for the `pydantic` field to hold them on each zone instance.

Reply:
Ah, hold on.  After re-reading your response, I double-checked and -- with the updated understanding from developing these changes -- I think I understand that these stats are already available in the `electricitymap.contrib.config.CO2EQ_PARAMETERS` object at runtime.  In particular, using a Python interactive shell appears to show the data under `CO2EQ_PARAMETERS['emissionFactors']['zoneOverrides']`.

That would seem to make sense, and could mean that I don't need the `ZoneEmissionFactors` model that I'd added here...

...and then the `del` statement can be restored too.